shopping-cart-client/src/app/services/login-services/login.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { Register } from 'src/app/register/register.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoginService {
+  setClosingFlag = new Subject<boolean>();
+  dataArray: Register[] = [];
+  isLoginFlag: boolean = false;
+  setLogout = new Subject<boolean>();
+  constructor() {}
+  saveRegisterationData(user: Register) {
+    if (localStorage.getItem('userData')) {
+      this.dataArray = JSON.parse(localStorage.getItem('userData') || '{}');
+    }
+    this.dataArray.push(user);
```

Comment:
Will pushing in an object work? Users still don't seem to be keyed on emails

Reply:
Yes, have changed the logic and pushing the object with mail ID as key